src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -421,6 +424,24 @@ public final class Guardrails implements GuardrailsMBean
                      format("The keyspace %s has a replication factor of %s, above the %s threshold of %s.",
                             what, value, isWarning ? "warning" : "failure", threshold));
 
+    public static final MaxThreshold maximumAllowableTimestamp =
+    new MaxThreshold("maximum_timestamp",
+                     null,
```

Comment:
The optional "reason" argument was added quite recently, so most guardrails don't use it. In fact, the only guardrails that use it are `allowFilteringEnabled` and `zeroTTLOnTWCSEnabled`. That argument is meant to provide the reason for having such a guardrail, so it can be included in the warn/fail messages. 

Maybe we could add here something like `Timestamps too far in the future are not realistic`, or something like that, if that makes sense. wdyt?